### ISSUE TITLE
A problem on training Re-Id heatmap with a vehicle dataset

### ISSUE THREAD

**Issue by rose-jinyang** [CaptainEven/FairMOTVehicle#1]
Hello
How are you?
Thanks for contributing this project.
I am going to train a model for tracking objects like cars.
I have a question about Re-Id.
In general, I think that there may be many "same cars" in a same position or different positions.
Here "same cars" means that they have same shape, color and brand although they are different objects.
In fact, I looked such cases in several vehicle datasets such as KITTI or DETRAC.
We assign different ids to these "same cars" to train Re-Id heatmap.
In this case, can the model for Re-Id converge?
I think that the model may converge in case to train persons rather than cars.
But what about training cars?
I doubt this point.
Thanks

**Comment by CaptainEven** [comment]
Hi, generally, different ids are assigned to different objects of cars even if they are the same in brand, color and shape, because they may still differs in recency, condition, year of manufacture, and details of decoration and so on. So, As long as the training data is enough and choose appropriate loss function(triplet loss, arc loss and so on) to capture fine-grained details, such policy is reasonable and can converge. Fined-grained classification is closely related to such Re-ID applications.
